### PR TITLE
Overload file serving methods with inline and filename params

### PR DESCRIPTION
**Pull request by mkurz**
We do have these overloads for other methods already.

This is my last pull request related to these file serving methods.

### CODE REVIEW

File: core/play/src/main/java/play/mvc/StatusHeader.java [hunk 1]
```diff
@@ -196,6 +196,37 @@ public Result sendBytes(byte[] content, Optional<String> contentType) {
     return new Result(status(), new HttpEntity.Strict(ByteString.fromArray(content), contentType));
   }
 
+  /**
+   * Send the given bytes.
+   *
+   * @param content The bytes to send.
+   * @param inline Whether it should be served as an inline file, or as an attachment.
+   * @param fileName The file name rendered in the {@code Content-Disposition} header.
```
Comment (ignasi35) [inline core/play/src/main/java/play/mvc/StatusHeader.java]:
```suggestion
   * @param fileName The file name rendered in the {@code Content-Disposition} header. The response will automatically include the MIME type deducing it from this fileName. 
```

File: core/play/src/main/java/play/mvc/StatusHeader.java [hunk 2]
```diff
@@ -217,6 +248,37 @@ public Result sendByteString(ByteString content, Optional<String> contentType) {
     return new Result(status(), new HttpEntity.Strict(content, contentType));
   }
 
+  /**
+   * Send the given ByteString.
+   *
+   * @param content The ByteString to send.
+   * @param inline Whether it should be served as an inline file, or as an attachment.
+   * @param fileName The file name rendered in the {@code Content-Disposition} header.
```
Comment (ignasi35) [inline core/play/src/main/java/play/mvc/StatusHeader.java]:
```suggestion
   * @param fileName The file name rendered in the {@code Content-Disposition} header. The response will automatically include the MIME type deducing it from this fileName. 
```

File: core/play/src/main/java/play/mvc/StatusHeader.java [hunk 3]
```diff
@@ -217,6 +248,37 @@ public Result sendByteString(ByteString content, Optional<String> contentType) {
...
+  public Result sendByteString(ByteString content, boolean inline, Optional<String> fileName) {
+    return sendByteString(content, inline, fileName, StaticFileMimeTypes.fileMimeTypes());
+  }
+
+  /**
+   * Send the given ByteString.
+   *
+   * @param content The ByteString to send.
+   * @param inline Whether it should be served as an inline file, or as an attachment.
+   * @param fileName The file name rendered in the {@code Content-Disposition} header.
```
Comment (ignasi35) [inline core/play/src/main/java/play/mvc/StatusHeader.java]:
```suggestion
   * @param fileName The file name rendered in the {@code Content-Disposition} header. The response will include the MIME type if <code>fileMimeTypes</code> includes it or fallback to <code>application/octet-stream</code> if unknown. 
```

File: core/play/src/main/scala/play/api/mvc/Results.scala [hunk 4]
```diff
@@ -643,15 +673,58 @@ trait Results {
...
       )
     }
+
+    /**
+     * Send an HTTP entity with this status.
+     *
+     * @param entity The entity to send.
+     * @param inline If the content should be rendered inline or as attachment.
+     * @param fileName The name of the resource, usually displayed in a file download dialog.
+     */
```
Comment (ignasi35) [inline core/play/src/main/scala/play/api/mvc/Results.scala]: Should this also doc the behavior of implicit `FileMimeTypes`? In the Java impl the overloads exposing `tileMimeTypes` on the signature do document the parameter.

**Comment by mkurz** [comment]
@ignasi35 Thanks for reviewing. I updated the javadoc as you suggested (just slightly different wording). Also mentioned it in the scaladoc.

**Pull request merged**